### PR TITLE
feat: push notification onboarding 

### PR DESCRIPTION
**Pull request by adriawh**
~~**Dependent on https://github.com/AtB-AS/mittatb-app/pull/4005 to be merged*~~

### 🔔 Added onboarding screen for push notifications

🆕 New  :
- Added screen to rootstack
- Added `@ATB_notification_permission_onboarded` to local storage to know if user has seen screen.
- Uses `enable_push_notifications` feature flag 
- Added "reset notification onboarding" in debug menu

🔨Changes:
- FinishedOnboarding() is now moved to the end of the login flow.

### CODE REVIEW

**Comment by adriawh** [comment]
@marius-at-atb Here you go

File: src/stacks-hierarchy/Root_NotificationPermissionScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const focusRef = useFocusOnLoad(true, 200);
+
+  const {completeNotificationPermissionOnboarding} = useAppState();
+
+  const {width: windowWidth} = useWindowDimensions();
+
+  const handleButtonPress = () => {
+    // TODO: add permission logic
+    completeNotificationPermissionOnboarding();
+    navigation.popToTop();
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/Root_NotificationPermissionScreen.tsx:29]: Also todo:
If the location permission screen will be triggered by the change from `completeNotificationPermissionOnboarding()`, then we might want to run `navigation.popToTop()` before `completeNotificationPermissionOnboarding()`

File: src/stacks-hierarchy/Root_NotificationPermissionScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView contentContainerStyle={styles.contentContainer}>
+        <View style={styles.mainView} ref={focusRef} accessible={true}>
+          <PushNotification width={windowWidth} />
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/Root_NotificationPermissionScreen.tsx:36]: Given the shape of the image and its positioning, I think it is better to set the size using height in this case

File: src/stacks-hierarchy/Root_TabNavigatorStack/Root_TabNavigatorStack.tsx [hunk 3]
```diff
@@ -48,8 +51,17 @@ export const Root_TabNavigatorStack = ({navigation}: Props) => {
       InteractionManager.runAfterInteractions(() =>
         navigation.navigate('Root_OnboardingStack'),
       );
+    } else if (!notificationPermissionOnboarded && pushNotificationsEnabled) {
```
Comment (marius-at-atb) [inline src/stacks-hierarchy/Root_TabNavigatorStack/Root_TabNavigatorStack.tsx:54]: Should only show if notification permission not already granted

File: src/translations/screens/NotificationPermission.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const NotificationPermissionTexts = {
+  title: _(
+    'Få varslinger før billetten din utløper',
+    'Get notifications before your ticket expires',
+    'Få varsel før billetten din går ut',
+  ),
+  description: _(
+    'Du kan tillate appen å sende deg varslinger for å få beskjed før billettene dine utløper.',
+    'You can allow the app to send you notifications to be informed before your tickets expire.',
+    'Du kan tillate appen å sende deg varsel for å bli informert før billettene dine går ut.',
```
Comment (marius-at-atb) [inline src/translations/screens/NotificationPermission.ts:12]:
```suggestion
    'Du kan tillate appen å sende deg varsel for å bli informert før billettane dine går ut.',
```

**Review by marius-at-atb — APPROVED** [review]
Great stuff!
Only some minor details in the comments

**Review by rosvik — COMMENTED** [review]
Not great to move the goal post after this is implemented, so not a blocker, but @oyvindmikkelsenatb just updated the sketches a bit and centered the content vertically:

Screenshot 2023-11-06 at 12 44 08

**Comment by marius-at-atb** [comment]
@rosvik I'm handling this with a new module in https://github.com/AtB-AS/kundevendt/issues/8622

**Review by rosvik — APPROVED** [review]
Looks good - approving since Marius has started to build on top of it as well 😄

**Pull request merged**